tests/blast/test_blast_input.h

```diff
@@ -0,0 +1,1166 @@
+        SEQAN_ASSERT_EQ(field5,      57u);
+    else
+        SEQAN_ASSERT_EQ(field5,      49u);
+    SEQAN_ASSERT_EQ(field6,          2u);
+    SEQAN_ASSERT_EQ(field7,          389u);
+    SEQAN_ASSERT_EQ(field8,          733u);
+    SEQAN_ASSERT_EQ(field9,          1u);
+    SEQAN_ASSERT_EQ(field10,         107u);
+
+    // read remaining matches
+    while (onMatch(it, BlastTabularLL()))
+        readMatch(it, BlastTabularLL(), field1); // only one field
+
+    // go to last record with matches
+    skipUntilMatch(it, BlastTabularLL());
+
+    // goto second (last) match
+    skipLine(it);
+
+    // check if exceptions are properly thrown
+    CharString exceptComment;
+    SEQAN_TRY
+    {
+        // no strings here to take the strings
+        readMatch(it, BlastTabularLL(), field3, field4, field5, field6, field7, field8, field9, field10, field11,
+                  field12);
+    } SEQAN_CATCH (BadLexicalCast const & e)
+    {
+        exceptComment = e.what();
+    }
+
+    SEQAN_ASSERT(startsWith(exceptComment, "Unable to convert"));
+    // skip rest of line
+    skipLine(it);
+
+    ifstream.close();
+}
+
+SEQAN_DEFINE_TEST(test_blast_read_match_lowlevel_tabular)
+{
+    // this testsBlastTabular#readMatch(),BlastTabular#skipMatch() and
+    //BlastTabular#skipUntilMatch
+    _test_blast_read_tabular_match_lowlevel(PLUS_HEADER_DEFAULTS);
+    // basic match reading should work between plus and legacy
+    _test_blast_read_tabular_match_lowlevel(LEGACY_HEADER_DEFAULTS);
+}
+
+inline void
+_test_blast_read_tabular_with_header(bool custom = false)
+{
+    // only used when defaults == false
+    typedef BlastMatchField<> TField;
+    std::array<typename TField::Enum, 13> customFields
+    {
+        {
+            TField::Enum::Q_SEQ_ID,
+            TField::Enum::S_SEQ_ID,
+            TField::Enum::LENGTH,
+            TField::Enum::N_IDENT,
+            TField::Enum::MISMATCH,
+            TField::Enum::POSITIVE,
+            TField::Enum::GAPS,
+            TField::Enum::Q_START,
+            TField::Enum::Q_END,
+            TField::Enum::S_START,
+            TField::Enum::S_END,
+            TField::Enum::FRAMES,
+            TField::Enum::BIT_SCORE
+        }
+    };
+
+    std::string inPath = std::string(SEQAN_PATH_TO_ROOT());
+    inPath += (custom ? PLUS_HEADER_CUSTOM : PLUS_HEADER_DEFAULTS);
+
+    std::ifstream ifstream(toCString(inPath),
+                           std::ios_base::in | std::ios_base::binary);
+    SEQAN_ASSERT(ifstream.is_open());
+
+    auto it = directionIterator(ifstream, Input());
+
+    BlastRecord<> r;
+    BlastIOContext<> context;
+
+    // read header of file
+    readHeader(context, it, BlastTabular());
+    SEQAN_ASSERT(getBlastTabularSpec(context) == BlastTabularSpec::HEADER);
+
+    // first line is record header
+    SEQAN_ASSERT(!onMatch(context, BlastTabular()));
+    SEQAN_ASSERT(onRecord(context, BlastTabular()));
+
+    readRecordHeader(r, it, context, BlastTabular());
```

Comment:
It was publicly documented before, so it is still public. I can change it to private, but I don't see whats wrong with the name. Every record has a header; since SeqAn defines the header to be file-global, I used RecordHeader to show that it is the header of the record. This is also explained in current dox.